client/lib/wp/browser.js

```diff
@@ -14,12 +15,16 @@ import wpcomSupport from 'lib/wp/support';
 let wpcom;
 
 if ( config.isEnabled( 'oauth' ) ) {
-	const oauthToken = require( 'lib/oauth-token' ),
-		requestHandler = require( 'lib/wpcom-xhr-wrapper' );
+	const oauthToken = require( 'lib/oauth-token' );
+	const requestHandler = config.isEnabled( 'sync-handler' )
```

Comment:
I would be for splitting it into `if {} else`
We have 4 different permutations here, so I would do 
```
if( config.isEnabled( 'sync-handler' ) ) {
const req... = ...
} else {
const....
}
```
and move it to line 16, then we dont have to duplicate the code.